src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3281,7 +3281,7 @@ ProtectionDomain protectionDomain() {
      * Return the Virtual Machine's Class object for the named
      * primitive type.
      */
-    static native Class<?> getPrimitiveClass(String name);
+    static native <T> Class<T> getPrimitiveClass(String name);
```

Comment:
I checked the code in java.base and this method is only called by the wrapper classes (and void), as expected.

I think the refactoring is good, but would like to see a comment here along the lines of "Use of type variable is acceptable since only used within java.lang," etc.

Reply:
Done. I added a comment "Type parameter T avoids redundant casts for trusted code." and removed "Virtual Machine's" as all Class objects are created by the JVM.